plugins/nf-validation/src/main/nextflow/validation/SamplesheetConverter.groovy

```diff
@@ -245,26 +246,34 @@ class SamplesheetConverter {
     }
 
     // Function to transform an input field from the samplesheet to its desired type
-    private static castToType(
-        String input,
+    private static castToNFType(
+        Object input,
         Map.Entry<String, Map> field
     ) {
         def String type = field['value']['type']
         def String key = field.key
 
+        if (type == "array") {
```

Comment:
Can we add a comment for our future selves? :)

Reply:
Documenting that this section just recurses the function if it's an array and that the array contents are still
limited to the other types supported?

Reply:
yes, please. Can be a very simple comment, just to know that this block is converting all elements of an array. Thanks!

Reply:
Updated with simple comment to explain logic